tests/Functional/SQL/ParserTest.php

```diff
@@ -17,4 +18,31 @@ public function testPostgreSQLJSONBQuestionOperator(): void
 
         self::assertTrue($this->connection->fetchOne('SELECT \'{"a":null}\'::jsonb ?? :key', ['key' => 'a']));
     }
+
+    public function testPostgresPlaceholders(): void
+    {
+        if (! TestUtil::isDriverOneOf('pgsql')) {
```

Comment:
It looks like this test passes on `pdo_pgsql` as well (not sure how it's implemented in PDO). Please move it into a dedicated test class and enable for `pdo_pgsql`.

Since it's a functional test, it would be better to name it in a way that it reflects the functionality being tested (something about PostgreSQL native positional parameters). The parser is an implementation detail, which isn't always relevant (specifically, in the case of `pdo_pgsql`).

Reply:
I moved it  to a separate class. Tried to enable it for pdo_pgsql as well, but it actually doesn't work, it doesn't seem to pass the values to the query and instead tries to insert null values.

Not sure if this is a limitation of pdo_pgsql or further changes would be required to Doctrine to support positional parameters for this driver. Error can be seen here https://github.com/debricked/dbal/actions/runs/12137000213/job/33839318280

Reply:
Seems to be a limitation of PDO https://www.php.net/pdo.prepare